cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,201 @@
+ * Computes a diff between current diff supplied as a Map and desired config supplied as a ConfigMap
+ *
+ */
+public class KafkaBrokerConfigurationDiff {
+
+    private static final Logger log = LogManager.getLogger(KafkaBrokerConfigurationDiff.class.getName());
+    private final Map<ConfigResource, Config> current;
+    private Collection<ConfigEntry> currentEntries;
+    private ConfigMap desired;
+    private KafkaConfiguration diff;
+    private KafkaVersion kafkaVersion;
+    private int brokerId;
+    private ArrayList<String> deletedEntries;
+
+    public static final Pattern IGNORABLE_PROPERTIES = Pattern.compile(
+            "^(broker\\.id"
+            + "|.*-909[1-4].ssl.keystore.location"
+            + "|.*-909[1-4]\\.ssl\\.keystore\\.password"
+            + "|.*-909[1-4]\\.ssl\\.keystore\\.type"
+            + "|.*-909[1-4]\\.ssl\\.truststore\\.location"
+            + "|.*-909[1-4]\\.ssl\\.truststore\\.password"
+            + "|.*-909[1-4]\\.ssl\\.truststore\\.type"
+            + "|.*-909[1-4]\\.ssl\\.client\\.auth"
+            + "|.*-909[1-4]\\.scram-sha-512\\.sasl\\.jaas\\.config"
+            + "|.*-909[1-4]\\.sasl\\.enabled\\.mechanisms"
+            + "|advertised\\.listeners"
+            + "|zookeeper\\.connect"
+            //+ "|log\\.dirs"*/
+            + "|super\\.users"
+            + "|broker\\.rack)$");
+
+    public KafkaBrokerConfigurationDiff(Map<ConfigResource, Config> current, ConfigMap desired, KafkaVersion kafkaVersion, int brokerId) {
+        this.current = current;
+        this.diff = emptyKafkaConf(); // init
+        this.desired = desired;
+        this.kafkaVersion = kafkaVersion;
+        this.brokerId = brokerId;
+        this.deletedEntries = new ArrayList<String>();
+        Config brokerConfigs = current.get(new ConfigResource(ConfigResource.Type.BROKER, Integer.toString(brokerId)));
+        if (brokerConfigs == null) {
+            throw new RuntimeException("Failed to get broker " + brokerId + " configuration");
+        }
+        this.currentEntries = brokerConfigs.entries();
+        this.diff = computeDiff();
+    }
+
+    private KafkaConfiguration emptyKafkaConf() {
+        Map<String, Object> conf = new HashMap<>();
+        return new KafkaConfiguration(conf.entrySet());
+    }
+
+    private HashMap<String, String> configMap2Map() {
+        HashMap<String, String> result = new HashMap<String, String>();
+        if (desired == null || desired.getData() == null || desired.getData().get("server.config") == null) {
+            return result;
+        }
+        String desireConf = desired.getData().get("server.config");
+
+        List<String> list = getLinesWithoutCommentsAndEmptyLines(desireConf);
+        for (String line: list) {
+            String[] split = line.split("=");
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
+        }
+        return result;
```

Comment:
Is this parsing a String in properties file format? Why not use `OrderedProperties`, or `Properties?